beginner_source/ax_multiobjective_nas_tutorial.py

```diff
@@ -0,0 +1,524 @@
+
+scheduler = Scheduler(
+    experiment=experiment,
+    generation_strategy=gs,
+    options=SchedulerOptions(
+        total_trials=total_trials, max_pending_trials=4
+    ),
+)
+
+
+######################################################################
+# Running the optimization
+# ------------------------
+#
+# Now that everything is configured, we can let Ax run the optimization
+# in a fully automated fashion. The Scheduler will periodially check
+# the logs for the status of all currenty running trials, and if a
+# trial completes the scheduler will update its status on the
+# experiment and fetch the observations needed for the Bayesian
+# optimization algorithm.
+#
+
+scheduler.run_all_trials()
+
+
+######################################################################
+# Evaluating the results
+# ----------------------
+#
+# We can now inspect the result of the optimization using helper
+# functions and visualizations included with Ax.
+
+######################################################################
+# First, we generate a dataframe with a summary of the results
+# of the experiment. Each row in this dataframe corresponds ot a
+# trial (i.e. a training job that was run), and contains information
+# on the status of the trial, the parameter configuration that was
+# evaluated, and the metric values that were observed. This provides
+# an easy way to sanity check the optimization.
+#
+
+from ax.service.utils.report_utils import exp_to_df
+
+df = exp_to_df(experiment)
+df.head(10)
+
+
+######################################################################
+# We can also visualize the Pareto frontier of tradeoffs between the
+# validation accuracy and the number of model parameters.
+#
+# .. tip::
+#   Ax uses Ploly to produce interactive plots, which allow you to
+#   do things like zoom, crop, or hover in order to view details
+#   of components of the plot. Try it out, and take a look at the
+#   `visualization tutorial <https://ax.dev/tutorials/visualizations.html>`__
+#   if you'd like to learn more).
+#
+# The final optimization results are shown in the Figure below where
+# the color corresponds to the iteration number for each trial.
+# We see that our method was able to successfully explore the
+# trade-offs and found both large models with high validation
+# accuracy as well as small models with compartively lower
+# validation accuracy.
+#
+
+from ax.service.utils.report_utils import _pareto_frontier_scatter_2d_plotly
+
+_pareto_frontier_scatter_2d_plotly(experiment)
+
+
+######################################################################
+# To better understand what our surrogate models have learned about
+# the black box objectives, we can take a look at the Leave-one-out
+# cross validation results. Since our models are Gaussian Processes,
+# they not only provide point predictions but also uncertainty estimates
+# about these predictions. A good model means
```

Comment:
```suggestion
# about these predictions. A good model means
```
Should "A good model means" be deleted?

Reply:
This was incomplete, thanks for catching this.

Reply:
```suggestion
# about these predictions. A good model means that the predicted means
# (the points in the figure) are close to the 45 degree line and that the
# confidence intervals cover the 45 degree line with the expected frequency
# (here we use 95% confidence intervals, so we would expect them to contain
# the true observation 95% of the time).
```